packages/@angular/cli/commands/new.ts

```diff
@@ -176,8 +176,8 @@ const NewCommand = Command.extend({
           }
         });
     } else {
-      createDirectory = mkdir(directoryName)
-        .catch(err => {
+      createDirectory = (mkdir as any)(directoryName)
```

Comment:
It seems that something is wrong with either `@types/node` or `@types/denodeify`. I Updated the versions but the error was still there. Added note to explain why it is being typed as `any` and a TODO for me to fix.